src/main/java/com/aizistral/nochatreports/mixins/both/MixinSerializer.java

```diff
@@ -0,0 +1,43 @@
+package com.aizistral.nochatreports.mixins.both;
+
+import com.aizistral.nochatreports.core.NoReportsConfig;
+import com.aizistral.nochatreports.core.NoReportServerStatus;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import net.minecraft.network.protocol.status.ServerStatus;
+import net.minecraft.util.GsonHelper;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import java.lang.reflect.Type;
+
+@Mixin(ServerStatus.Serializer.class)
+public class MixinSerializer {
+
+	@Inject(method = "serialize(Lnet/minecraft/network/protocol/status/ServerStatus;" +
+			"Ljava/lang/reflect/Type;Lcom/google/gson/JsonSerializationContext;)Lcom/google/gson/JsonElement;",
+			at = @At("RETURN"))
+	private void onSerialize(ServerStatus serverStatus, Type type, JsonSerializationContext context,
+			CallbackInfoReturnable<JsonElement> info) {
+		if (NoReportsConfig.includeQueryData() && serverStatus instanceof NoReportServerStatus noReportStatus) {
+			((JsonObject)info.getReturnValue()).addProperty("saveMinecraft", noReportStatus.hasChatReporting());
```

Comment:
Why the property is called `saveMinecraft` while method is `hasChatReporting`. Logically, shouldn't `hasChatReporting` return true only if server doesn't, in fact, have NCR installed?

Reply:
So ill rename the method to `hasNoChatReports`